commit b4a22303b517ebd026515746ff3ac4faa2d93c6c
Author: David Faure <faure@kde.org>
Date:   Thu Aug 13 12:33:54 2015 +0200

    split out the factory, to fix compilation of the unittest for some people
    
    Change-Id: I46296eab3f3d5130544a9858942339e6654d8123
diff -Naur kio-5.13.0/src/kpasswdserver/CMakeLists.txt kio-5.13.0.tpg/src/kpasswdserver/CMakeLists.txt
--- kio-5.13.0/src/kpasswdserver/CMakeLists.txt	2015-08-04 11:03:40.000000000 +0000
+++ kio-5.13.0.tpg/src/kpasswdserver/CMakeLists.txt	2015-08-13 22:16:53.370369389 +0000
@@ -2,7 +2,7 @@
 
 add_subdirectory(tests)
 
-set(kded_kpasswdserver_SRCS kpasswdserver.cpp )
+set(kiod_kpasswdserver_SRCS kpasswdserver.cpp kiod_kpasswdserver.cpp)
 
 qt5_add_dbus_adaptor(kded_kpasswdserver_SRCS org.kde.KPasswdServer.xml kpasswdserver.h KPasswdServer)
 
diff -Naur kio-5.13.0/src/kpasswdserver/kiod_kpasswdserver.cpp kio-5.13.0.tpg/src/kpasswdserver/kiod_kpasswdserver.cpp
--- kio-5.13.0/src/kpasswdserver/kiod_kpasswdserver.cpp	1970-01-01 00:00:00.000000000 +0000
+++ kio-5.13.0.tpg/src/kpasswdserver/kiod_kpasswdserver.cpp	2015-08-13 22:15:52.172675718 +0000
@@ -0,0 +1,9 @@
+#include "kpasswdserver.h"
+
+#include <kpluginfactory.h>
+#include <kpluginloader.h>
+
+K_PLUGIN_FACTORY_WITH_JSON(KPasswdServerFactory, "kpasswdserver.json", registerPlugin<KPasswdServer>();)
+
+#include "kiod_kpasswdserver.moc"
+
diff -Naur kio-5.13.0/src/kpasswdserver/kpasswdserver.cpp kio-5.13.0.tpg/src/kpasswdserver/kpasswdserver.cpp
--- kio-5.13.0/src/kpasswdserver/kpasswdserver.cpp	2015-08-04 11:03:40.000000000 +0000
+++ kio-5.13.0.tpg/src/kpasswdserver/kpasswdserver.cpp	2015-08-13 22:17:48.640935645 +0000
@@ -1,3 +1,4 @@
+
 /*
     This file is part of the KDE Password Server
 
@@ -34,9 +35,6 @@
 #include <kusertimestamp.h>
 #include <klocalizedstring.h>
 
-#include <kpluginfactory.h>
-#include <kpluginloader.h>
-
 #include <QPushButton>
 #include <QTimer>
 #include <ctime>
@@ -44,8 +42,6 @@
 static QLoggingCategory category("org.kde.kio.kpasswdserver");
 
 
-K_PLUGIN_FACTORY_WITH_JSON(KPasswdServerFactory, "kpasswdserver.json", registerPlugin<KPasswdServer>();)
-
 #define AUTHINFO_EXTRAFIELD_DOMAIN QLatin1String("domain")
 #define AUTHINFO_EXTRAFIELD_ANONYMOUS QLatin1String("anonymous")
 #define AUTHINFO_EXTRAFIELD_BYPASS_CACHE_AND_KWALLET QLatin1String("bypass-cache-and-kwallet")
@@ -1099,4 +1095,3 @@
 }
 
 
-#include "kpasswdserver.moc"
